File: java/java-practise/src/main/java/day/six/java-cucumber-healthcare/src/test/java/bdd/steps/ChatbotAnswerSteps.java
package bdd.steps;

import domain.SupportChatbot;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import static org.assertj.core.api.Assertions.assertThat;

public class ChatbotAnswerSteps {
    private SupportChatbot chatbot;
    private String chatbotAnswer;

    @Given(
            "the clinic support chatbot is available"
    )
    public void theClinicSupportChatbotIsAvailable() {
        chatbot = new SupportChatbot();
    }

    @When("I ask the chatbot {string}")
    public void iAskTheChatbot(String question) {
        chatbotAnswer = chatbot.answer(question);
    }

    @Then(
            "the chatbot answer should contain {string}"
    )
    public void theChatbotAnswerShouldContain(
            String expectedText
    ) {
        assertThat(chatbotAnswer)
                .contains(expectedText);
    }
}

File: scharp/day/six/CSharpSpecFlowHealthcareExample/Bindings/ChatbotAnswerSteps.cs

using CSharpSpecFlowHealthcareExample.Domain;
using NUnit.Framework;
using TechTalk.SpecFlow;

namespace CSharpSpecFlowHealthcareExample.Bindings
{
    [Binding]
    public sealed class ChatbotAnswerSteps
    {
        private SupportChatbot _chatbot = null!;

        private string _chatbotAnswer = string.Empty;

        [Given(
            @"^the clinic support chatbot is available$")]
        public void GivenTheClinicSupportChatbotIsAvailable()
        {
            _chatbot = new SupportChatbot();
        }

        [When(@"^I ask the chatbot ""([^""]*)""$")]
        public void WhenIAskTheChatbot(
            string question)
        {
            _chatbotAnswer =
                _chatbot.Answer(question);
        }

        [Then(
            @"^the chatbot answer should contain ""([^""]*)""$")]
        public void ThenTheChatbotAnswerShouldContain(
            string expectedText)
        {
            Assert.That(
                _chatbotAnswer,
                Does.Contain(expectedText));
        }
    }
}
